File: Advent of Code/Codepal/Python/Advent1.py
from itertools import combinations

def count_combinations(num_containers, capacities):
    """
    This function takes the number of available containers and their capacities as arguments and returns the number of
    different combinations of containers that can exactly fit all the eggnog.
    
    Parameters:
    num_containers (int): The number of available containers
    capacities (list): A list of integers indicating the capacities of the containers
    
    Returns:
    int: The number of different combinations of containers that can exactly fit all the eggnog
    """
    try:
        # Check if the number of containers is an integer
        if not isinstance(num_containers, int):
            raise TypeError("The number of containers must be an integer")
        
        # Check if all capacities are integers
        if not all(isinstance(capacity, int) for capacity in capacities):
            raise TypeError("All capacities must be integers")
        
        # Calculate all possible combinations of containers
        container_combinations = []
        for i in range(1, num_containers+1):
            container_combinations += combinations(capacities, i)
        
        # Count the number of combinations that can fit all the eggnog
        count = 0
        for combination in container_combinations:
            if sum(combination) == 150:
                count += 1
        
        return count
    except TypeError as e:
        # Log the error
        print(f"Error: {e}")
        return 0

File: Advent of Code/GithubCopilot/C/Advent1.c
#include <stdio.h>
#include <math.h>
//The elves bought too much eggnog again - 150 liters this time. To fit it all into your refrigerator, you'll need to move it into smaller containers. You take an inventory of the capacities of the available containers. For example, suppose you have containers of size 20, 15, 10, 5, and 5 liters. If you need to store 25 liters, there are four ways to do it:
//15 and 10
//20 and 5 (the first 5)
//20 and 5 (the second 5)
//15, 5, and 5
//Filling all containers entirely, how many different combinations of containers can exactly fit all 150 liters of eggnog?

//The input data is given through standard input on the following format: The first row contains the number of available containers Each following line contains the capacity of a container. All capacities are integers and indicate liters. 

//The output data should be printed to standard output given in the following format: One line with the answer. 

int power(int base, int exponent) {
    int i, result;
    result = 1;
    for (i = 0; i < exponent; i++) {
        result *= base;
    }
    return result;
}

int main() {
	int i, j, k, l, m, n, o, p, q, r, s, t, u, v, z, count, temp;
	int input[20];
	int output[20];
	int numContainers, sum, numSolutions;
	
	numSolutions = 0;
	sum = 150;
	
	for (i = 0; i < 20; i++) {
		input[i] = 0;
		output[i] = 0;
	}
	
	scanf("%d", &numContainers);
	
	for (i = 0; i < numContainers; i++) {
		scanf("%d", &input[i]);
	}
    int powerCap = power(2, numContainers);
	
	for (i = 0; i < powerCap; i++) {
		temp = i;
		for (j = 0; j < numContainers; j++) {
			output[j] = temp % 2;
			temp = temp / 2;
		}
		sum = 0;
		for (k = 0; k < numContainers; k++) {
			sum += input[k] * output[k];
		}
		if (sum == 150) {
			numSolutions++;
		}
	}
	
	printf("%d", numSolutions);
	
	return 0;
}